### PR TITLE
III-6310 Handle update with empty description

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added new exception `DescriptionShouldNotBeEmpty`

### Changed
- Handle an empty description and dispatch a `DeleteDescription` for this

---

Ticket: https://jira.uitdatabank.be/browse/III-6310

### CODE REVIEW

File: src/Http/Offer/UpdateDescriptionRequestHandler.php [hunk 1]
```diff
@@ -47,26 +50,35 @@ public function handle(ServerRequestInterface $request): ResponseInterface
...
 
-        if ($offerType->sameAs(OfferType::event())) {
-            $updateDescription = new EventUpdateDescription(
-                $offerId,
-                $language,
-                $description
-            );
-        } else {
-            $updateDescription = new PlaceUpdateDescription(
+            if ($offerType->sameAs(OfferType::event())) {
```
Comment (grubolsch) [inline src/Http/Offer/UpdateDescriptionRequestHandler.php:59]: Can we make either a seperate function or class helper that expects any type of UpdateDescription and just returns the correct type of Command?

Comment (grubolsch) [reply src/Http/Offer/UpdateDescriptionRequestHandler.php:59]: Would be useful for example to have an OfferUpdateDescription class with a static method "buildCommand()" that returns the correct object? 
If you think that is overkill I am also happy with just a function in this handler.

**Review by grubolsch — CHANGES_REQUESTED** [review]

File: src/Http/Offer/UpdateDescriptionRequestHandler.php (hunk 1 above)
Comment (LucWollants) [reply src/Http/Offer/UpdateDescriptionRequestHandler.php:59]: The approach described here should be used https://jira.publiq.be/browse/III-3834. This is not inside the scope of this refactor of value objects and that is why I didn't change it.

**Review by grubolsch — APPROVED** [review]

File: src/Http/Offer/UpdateDescriptionRequestHandler.php (hunk 1 above)
Comment (grubolsch) [reply src/Http/Offer/UpdateDescriptionRequestHandler.php:59]: I will approve then

**Pull request merged**